backend/device-management-main/src/main/java/com/amigoscode/devicemanagement/external/storage/devicesetting/DeviceSettingEntity.java

```diff
@@ -0,0 +1,53 @@
+package com.amigoscode.devicemanagement.external.storage.devicesetting;
+
+import lombok.*;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.annotation.TypeAlias;
+import org.springframework.data.mongodb.core.index.Indexed;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.time.ZonedDateTime;
+
+@Document("DeviceSettings")
+@TypeAlias("DeviceSettingEntity")
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
+class DeviceSettingEntity {
+
+    @Id
+    private String  id;
+
+    @Indexed(unique = true)
+    private String deviceId;
+
+    private Integer measurementPeriod;
+    private Boolean isMeasurementEnabled;
+    private ZonedDateTime createdAt;
+    private ZonedDateTime deletedAt;
+    private ZonedDateTime updatedAt;
+    private String updatedBy;
+
+
+    public DeviceSettingEntity(String deviceId, Integer measurementPeriod, Boolean isMeasurementEnabled) {
+        this.deviceId = deviceId;
+        this.measurementPeriod = measurementPeriod;
+        this.isMeasurementEnabled = isMeasurementEnabled;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        DeviceSettingEntity that = (DeviceSettingEntity) o;
+        return id.equals(that.id);
+    }
+
+    @Override
+    public int hashCode() {
+        return 0;
```

Comment:
why 0?

Reply:
We relied on this article:
https://thorben-janssen.com/ultimate-guide-to-implementing-equals-and-hashcode-with-hibernate/

In our case, MongoDB assigns the id. Therefore, we followed the suggestions:

“If you tell Hibernate to generate your primary key values, you need to use a fixed hash code, and your equals() method requires explicit handling of null values.”

Reply:
good to know